include/envoy/http/header_map.h

```diff
@@ -294,7 +294,8 @@ class HeaderEntry {
   HEADER_FUNC(Protocol)                                                                            \
   HEADER_FUNC(Scheme)                                                                              \
   HEADER_FUNC(TE)                                                                                  \
-  HEADER_FUNC(UserAgent)
+  HEADER_FUNC(UserAgent)                                                                           \
+  HEADER_FUNC(Cookie)
```

Comment:
There seems to be no set-cookie related changes in this PR.  Is the set-cookie behavior already correct and just requires extra testing?

Reply:
I think this will result in Envoy inlining cookie headers, which is a pretty large change we should avoid if we can.